client/src/components/user/UserRouter.js

```diff
@@ -44,7 +44,7 @@ const UserRouter = () => {
           });
     };
     checkLoggedIn();
-  }, []);
+  });
```

Comment:
The empty array argument is intended, causing the useEffect callback to execute once, like componentDidMount. By deleting it, you broke the "Log Out" functionality.